.github/workflows/update-dependencies.yml

```diff
@@ -0,0 +1,62 @@
+name: Update dependencies
+on:
+  schedule:
+    # Run automatically at 7AM PST Tuesday
+    - cron: '0 14 * * 2'
+  workflow_dispatch:
+
+jobs:
+  update-dependencies-and-test:
+    name: Update Dependencies & Test
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    strategy:
+      matrix:
+        java: [ 8, 11 ]
+
+    steps:
+      - name: Checkout sendgrid-java
+        uses: actions/checkout@v2
+
+      - name: Updating semver dependencies
+        run: make update-deps
+
+      - name: Run Unit Tests
+        run: make test-docker version=${{ matrix.java }}
+
+  commit-dependencies:
+    name: Commit Updated Dependencies
+    if: success()
+    needs: [ update-dependencies-and-test ]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sendgrid-java
+        uses: actions/checkout@v2
+
+      - name: Updating semver dependencies
+        run: make update-deps
```

Comment:
Why run these step again instead of adding the Add & Commit step to the end of the "update-dependencies-and-test" job?

Reply:
I separated the commit into a separate job because the `update-dependencies-and-test` runs on the java 8 and 11. If the commit is included in the `update-dependencies-and-test` job tries to commit the changes to the pom file twice.

Reply:
Can't we just add an `if:` condition to the step? Like we do for sonarcloud scanning in twilio (https://github.com/twilio/twilio-python/blob/main/.github/workflows/test-and-deploy.yml#L43)

Reply:
The issue is what if 8 and 17 pass but 11 fails tests for some non-transient issue. Then 17 would commit the change (assuming the `if` is for 17) but the tests would be failing for 11.

May be able to limit job concurrency so only 1 job can run at a time: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#concurrency

Note that the order of the matrix would then be crucial as we'd only want to commit on the last matrix value.

> The order that you define a matrix matters. The first option you define will be the first job that runs in your workflow.

Reply:
I ended up using [max-parallel](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategymax-parallel) to force the matrix to run 1 version at a time. After a the first language version successfully passes the next language version job runs.